### PR TITLE
Fix#980. Fixed hardcode colors. 

### LINKED ISSUE #980: Fix hardcoded CSS variables in VaInput, VaSelect
https://localhost:3000/admin/dashboard?theme=dark

image
The dropdown must use some background CSS variables as input does.

image
Must use something like this `setHSLAColor(getColor('success'), { a: 0.3 })`

image
Checkbox and Radio should also be relative to `--va-background` color.

image

image
Must use auto text color

image
Must use auto text color like badge do

image

image
Modals must have a dark background

image
Divider color

----

There can be issues with theme color instead of unrelative CSS variables. 

You can implement a theme switcher (take it from Vuestic-Admin) for docs.

### PR DESCRIPTION
**Pull request by Sadnessa**
## Description
Closes #980.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Improvement/refactoring (non-breaking change that doesn't add any feature but make things better)

### CODE REVIEW

**Review by m0ksem — APPROVED** [review]
Looks like we need addtional colors in getTextColor: `text-dark, text-light` instead of `white, dark`.
We also need to replace `-va-white, -va-dark, -va-divider, -va-background` to `-va-background, -va-backgorund-mute, -va-background-soft`
I took this color names from vue sample project (made with vue init) here:
image



Here I just change -va-white, -va-dark and -va-background colors to get black theme. I also added text colors.
image

image


Colors should be reimagined in order of https://github.com/epicmaxco/vuestic-ui/issues/744

Need to deal with box-shadows. They are usually hardcoded. Must be another task.
image

**Comment by m0ksem** [comment]
@aluarius, I suggest changing getTextColor to 
```ts
export const getTextColor = (color: ColorInput, darkColor = 'var(--va-text-dark)', lightColor = 'var(--va-text-light)') => {
  const { R, G, B } = new ColorTranslator(color)
  const isLightBackground = Math.sqrt(R * R * 0.241 + G * G * 0.691 + B * B * 0.068) > 130
  return isLightBackground ? darkColor : lightColor
}
```

Then you are able to check dark theme update packages/docs/src/config/theme-config.ts:

```ts
export const COLOR_THEMES: Record<ThemeName, Record<string, string>> = {
  [ThemeName.DEFAULT]: {
    primary: '#2C82E0',
    secondary: '#767C88',
    background: '#0d1117',
    success: '#40e583',
    info: '#2c82e0',
    danger: '#DE1041',
    warning: '#FFAC0A',
    gray: '#babfc2',
    white: '#1B1A1F',
    dark: '#ffffff',
    divided: '#000',
    'text-dark': '#000',
    'text-light': '#fff',
  },
```

File: packages/ui/src/services/color-config/color-config.ts [hunk 1]
```diff
@@ -51,6 +51,10 @@ export const getColor = (prop?: string, defaultColor: string = DEFAULT_COLOR, pr
     return prop
   }
 
+  if (preferVariables && isCSSVariable(prop)) {
```
Comment (aluarius) [inline packages/ui/src/services/color-config/color-config.ts:54]: @m0ksem won't be here again a SSR problem because of `isCSSVariable`?

**Review by aluarius — APPROVED** [review]

File: packages/ui/src/services/color-config/color-config.ts (hunk 1 above)
Comment (m0ksem) [reply packages/ui/src/services/color-config/color-config.ts:54]: `isCSSVariable` uses regex under the hood. CSS variable available in SSR with `renderCSSVariable`. Can't see any problems with SSR.

File: packages/ui/src/components/va-data-table/VaDataTable.vue [hunk 2]
```diff
@@ -534,8 +534,11 @@ export default defineComponent({
     }
 
     .va-data-table__table-tr {
+      position: relative;
+      z-index: 0;
+
       &.selected {
-        background-color: var(--selected-color);
+        background-color: var(--va-selected-color);
```
[the remark this answers is not in the document]
Comment (Sadnessa) [reply packages/ui/src/components/va-data-table/VaDataTable.vue:541]: This variable used in useStyleable hook. There are a lot of variables without '--va' prefix. Should we add this prefix to all css variables in useStyleable hook?

[the remark this answers is not in the document]
Comment (RVitaly1978) [reply packages/ui/src/components/va-data-table/VaDataTable.vue:541]: Yes. That would be right.

File: packages/ui/src/components/va-data-table/VaDataTable.vue [hunk 3]
```diff
@@ -533,25 +533,25 @@ export default defineComponent({
       }
     }
 
-    .va-data-table__table-tr {
-      position: relative;
-      z-index: 0;
-
-      &.selected {
-        background-color: var(--va-selected-color);
-      }
```
Comment (RVitaly1978) [inline packages/ui/src/components/va-data-table/VaDataTable.vue:539]: Table rows can be `selected` not only when table is `striped`. Please don't remove this from here.

**Review by RVitaly1978 — APPROVED** [review]

**Pull request merged**